src/components/screens/timebank-view-all-screen.tsx

```diff
@@ -0,0 +1,302 @@
+  const [persons] = useAtom(personsAtom);
+  const [loading, setLoading] = useState(false);
+  const [searchInput, setSearchInput] = useState("");
+  const [displayedPersonsTotalTime, setDisplayedPersonsTotalTime] = useState<PersonWithTotalTime[]>([]);
+
+  /**
+   * Renders the customized tooltip for charts
+   *
+   * @param props props of the custom tooltip
+   */
+  const renderCustomizedTooltip = (props: TooltipProps<ValueType, NameType>) => {
+    const { active, payload } = props;
+
+    if (!active || !payload || !payload.length) {
+      return null;
+    }
+
+    const selectedData = payload[0];
+
+    if (!selectedData.value || !selectedData.name) {
+      return null;
+    }
+
+    const sectionName = {
+      [WorkTimeCategory.BILLABLE_PROJECT]: strings.timebank.billableProjectTime,
+      [WorkTimeCategory.NON_BILLABLE_PROJECT]: strings.timebank.nonBillableProjectTime,
+      [WorkTimeCategory.INTERNAL]: strings.timebank.internalTime
+    }[selectedData.name];
+
+    return (
+      <Box style={{ backgroundColor: "rgba(0, 0, 0)", opacity: "70%" }}>
+        <Typography
+          variant="h6"
+          style={{
+            color: "#fff",
+            padding: theme.spacing(1)
+          }}
+        >
+          {`${sectionName}: ${getHoursAndMinutes(selectedData.value as number)}`}
+        </Typography>
+      </Box>
+    );
+  };
+
+  /**
+   * Render piechart component
+   *
+   * @param personWithTotalTime person with total time object
+   * @param legend legend boolean
+   */
+  const renderPieChart = (personWithTotalTime: PersonWithTotalTime, legend: boolean) => {
+    const { person, personTotalTime } = personWithTotalTime;
+
+    if (!person || !personTotalTime) return null;
+
+    const workTimeData: WorkTimeTotalData[] = [
+      { name: WorkTimeCategory.BILLABLE_PROJECT, balance: personTotalTime.billableProjectTime },
+      {
+        name: WorkTimeCategory.NON_BILLABLE_PROJECT,
+        balance: personTotalTime.nonBillableProjectTime
+      },
+      { name: WorkTimeCategory.INTERNAL, balance: personTotalTime.internalTime }
+    ];
+
+    return (
+      <ResponsiveContainer width={"75%"} height={150}>
+        <PieChart>
+          <Pie
+            cx="50%"
+            cy="50%"
+            dataKey="balance"
+            data={workTimeData}
+            label={(props) => getHoursAndMinutes(props.value)}
+          >
+            {workTimeData.map((_entry, index) => (
+              <Cell key={`cell-${index}`} fill={COLORS[index % COLORS.length]} />
+            ))}
+          </Pie>
+          {legend ? <Legend wrapperStyle={{ position: "relative" }} /> : null}
+          <Tooltip content={renderCustomizedTooltip} />
+        </PieChart>
+      </ResponsiveContainer>
+    );
+  };
+
+  /**
+   * Populate one person's total work data
+   *
+   * @param person person with total time data
+   */
+  const populatePersonTotalTimeData = async (
```

Comment:
This does not actually populate anything, rename the method to reflect what it actually does